### ISSUE TITLE
Blender 4.0.2 - Crashes after starting and launcher disappears from Blender's interface **[BUG]**

### ISSUE THREAD

**Issue by SentinelForce** [CGCookie/retopoflow#1293]
**PLEASE READ THE FOLLOWING AND USE THE TEMPLATE TO POST YOUR ISSUE**

*Questions not directly related to the RetopoFlow project are considered general and should be posted elsewhere (e.g. git, blender, python, etc).*

**Before raising the issue, please check the following first:**

- [x] Read through the README.md file in the branch used to assure the process taken is correct
- [x] Check the existing [issues](https://github.com/CGCookie/retopoflow/issues) and [pull requests](https://github.com/CGCookie/retopoflow/pulls) to make sure the issue has not already been reported and/or fixed.
- [x] You can replicate the bug and will provide as much info as possible with blend files, screenshots, logfiles, as well as show the expected result and actual result.

&nbsp;

Issue Template
-----------------------------------

---
Please prefix your issue name with one of the following: **[BUG]** **[PROPOSAL]** **[QUESTION]**

Retopoflow Version: 3.4.2

Blender Version/Hash: 4.0.2

Platform Version/Distribution: Windows 11

&nbsp;

**Issue:**

After entering into the Retopo workspace and using for a very short ammount of time, Retopoflow will randomly crash and boot me back into blender but the launcher in the menus will disappear and not be useable until I relaunch Blender. There is no crash log that saves to blender to upload when the crash happens. I only get a "retopo flow has encountered an unexpected error"

![Screenshot 2024-01-03as 164853](https://github.com/CGCookie/retopoflow/assets/13028408/fafbdf30-d349-457f-816e-6bcb2fca3ef3)
![Screenshot 2024-01-03 164542](https://github.com/CGCookie/retopoflow/assets/13028408/cf517038-351c-40f7-af96-2dbb8e6e2372)

&nbsp;

**How to Reproduce:**

- Launch Blender
- Open Retopoflow
- Use for 1-2 minutes

{*attach any files needed to replicate the issue*}

**Comment by vxlcoder** [comment]
thanks for the report, @SentinelForce.

I have a few follow-up questions to try to debug this:

- Has RetopoFlow worked before, and only recently it hasn't?
- Where did you get the zip of RF to install?  Blender Market, cloned RF repo from GitHub, downloaded zip from RF's main GH page, download zip from releases page, etc.
- Does it crash when you use RF on the default scene with default cube?

Can you try starting RF, then after it crashes, copy-paste the output from window's system console to here?  There might be a lot printed there, so make sure to copy everything.  please see [link](https://docs.retopoflow.com/debugging.html#windows) for more details.

Thanks for your help!  we'll get you back up and retopo'ing soon

**Comment by SentinelForce** [comment]
> Has RetopoFlow worked before, and only recently it hasn't?

> Where did you get the zip of RF to install? Blender Market, cloned RF repo from GitHub, downloaded zip from RF's main GH page, download zip from releases page, etc.

> Does` it crash when you use RF on the default scene with default cube?

Retopo did work before I updated. I hadn't updated in a while and the old version I still had installed would no longer appear.

The zip for the addon I got directly from the Blender Market downloads

Doesn't seem to crash on a default cube so that might be something. Doesn't seem to misbehave with other blender files either?

I also opened the system console, like in the link but when the crashed happened nothing got recorded?

> Addon Common: use_bgl_default=False use_gpu_default=True use_gpu_scissor=True
Addon Common: {'backend': 'OPENGL', 'device': 'NVIDIA', 'vendor': 'NVIDIA Corporation', 'renderer': 'NVIDIA GeForce RTX 3070/PCIe/SSE2', 'version': '4.6.0 NVIDIA 536.99', 'compute_shader_support_get': True, 'hdr_support_get': False, 'max_batch_indices_get': 1048576, 'max_batch_vertices_get': 1048576, 'max_texture_layers_get': 2048, 'max_texture_size_get': 32768, 'max_textures_frag_get': 32, 'max_textures_geom_get': 32, 'max_textures_get': 192, 'max_textures_vert_get': 32, 'max_uniforms_frag_get': 4096, 'max_uniforms_vert_get': 4096, 'max_varying_floats_get': 124, 'max_vertex_attribs_get': 16, 'shader_image_load_store_support_get': True}
RetopoFlow: Initial load
RetopoFlow git: None
RetopoFlow options path: C:\Users\Jacob\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\RetopoFlow\RetopoFlow_options.json
cgcookie addon: Verbose is enabled
cgcookie addon: Read in JSON settings from file
Warning: addon-module bone_selection_sets found module but without '__addon_enabled__' field, possible name collision from file: 'E:\\SteamLibrary\\steamapps\\common\\Blender\\4.0\\scripts\\addons\\bone_selection_sets.py'
Exception in module register(): C:\Users\Jacob\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\RenderManForBlender\__init__.py
ID 1704286567973
tmp C:\Users\Jacob/AppData/Roaming/Xolotl/splink/tmp
prefs C:\Users\Jacob/AppData/Roaming/Xolotl/splink/prefs
blenderkit addon: Running background update callback
blenderkit addon: Attempted popup prompt
blenderkit addon: Running the install popup handler.
Megascans Plugin Error initializing the thread. Error:  [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted
ID 1704286570443
tmp C:\Users\Jacob/AppData/Roaming/Xolotl/splink/tmp
prefs C:\Users\Jacob/AppData/Roaming/Xolotl/splink/prefs

**Comment by vxlcoder** [comment]
thanks for the details.

We have found that getting RF from non-official places has caused problems.  The Blender Market zip should be just fine, though.

you said that it worked before you updated.  what did you update: blender or RF (or both?)?  if you updated RF, did you simply install over the previous version?  maybe try completely removing previous version, restart Blender, then install RF again?

you're right. i don't see any issues related to RF in the console log.  There seems to be quite a few issues with other add-ons, though: `bone_selection_sets`, `RenderManForBlender`, `Megascans`.  maybe try disabling them to see if RF still crashes?

**Comment by SentinelForce** [comment]
I updated both, Blender to 4.0.2 and RF to most recent today. I uninstalled the previous then installed the latest.

The addons above they shouldn't be causing a problem I don't know why they're getting flagged? megascans isn't installed so don't know why it's showing up? old version? and bone selection isn't activated.

**Comment by vxlcoder** [comment]
I agree that the other addons shouldn't cause an issue, but I thought I'd make the suggestion as this is an issue I haven't seen before, and those items stuck out to me :)

If the issue seems only to happen to that one .blend file you screenshotted above, there might be something peculiar with that file.

would you be able to send it to us at retopoflow@cgcookie.com so we can try to reproduce this error here?

again, thank you for the info and helping us debug this issue!

**Comment by SentinelForce** [comment]
sent it through

**Comment by jlampel** [comment]
Closing this out as it appears to be handled over email, but please feel free to re-open if it was not resolved and is still an issue in Retopoflow 3. Since we no longer have the same Retopoflow mode, this is not an issue in Retopoflow 4.
